### PR TITLE
Add distillation training recipe for CorrDiff

### PR DESCRIPTION
**Pull request by jialusui1102**
# PhysicsNeMo Pull Request

## Description

Added distillation recipe and helper functions for CorrDiff distillation training with fastgen framework

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: examples/weather/corrdiff/distill.py [hunk 1]
```diff
@@ -0,0 +1,1149 @@
...
+                                                loss_valid,
+                                                loss_map_valid,
+                                                output_batch_valid,
+                                            ) = loss_fn(**loss_valid_kwargs)
+
+                                        loss_valid = (
+                                            (loss_valid.sum() / batch_size_per_gpu)
+                                            .cpu()
+                                            .item()
+                                        )
```
[the remark this answers is not in the document]
Comment (jialusui1102) [reply examples/weather/corrdiff/distill.py:948]: skip since that's the original code

File: examples/weather/corrdiff/distill.py [hunk 2]
```diff
@@ -0,0 +1,1149 @@
...
+    handle_and_clip_gradients,
+    is_time_for_periodic_task,
+)
+
+torch._dynamo.reset()
+# Increase the cache size limit
+torch._dynamo.config.cache_size_limit = 264  # Set to a higher value
+torch._dynamo.config.verbose = True  # Enable verbose logging
+torch._dynamo.config.suppress_errors = False  # Forces the error to show all details
+torch._logging.set_logs(recompiles=True, graph_breaks=True)
```
[the remark this answers is not in the document]
Comment (jialusui1102) [reply examples/weather/corrdiff/distill.py:83]: Keep it there for torch.compile

File: examples/weather/corrdiff/helpers/distill_helpers.py [hunk 3]
```diff
@@ -0,0 +1,898 @@
...
+        self.net, self.ema = build(self.config, use_ema=self.use_ema)
+
+        # instantiate the teacher and consistency network
+        if self.config.loss_config.use_cd:
+            self.teacher = deepcopy(self.net)
+            self.teacher.eval().requires_grad_(False)
+
+
+class SCMModel(SCMBaseModel):
+    """-time Consistency Model with TrigFlow for CorrDiff distillation.
```
[the remark this answers is not in the document]
Comment (jialusui1102) [reply examples/weather/corrdiff/helpers/distill_helpers.py]: updated

File: examples/weather/corrdiff/generate.py [hunk 4]
```diff
@@ -197,6 +199,19 @@ def main(cfg: DictConfig) -> None:
...
+        )
+        sampler_fn = partial(
+            few_step_sampler,
+            sigma_max=sigma_max,
+            sigma_mid=sigma_mid,
+            patching=patching,
+        )
+
     else:
         raise ValueError(f"Unknown sampling method {cfg.sampling.type}")
```
[the remark this answers is not in the document]
Comment (jialusui1102) [reply examples/weather/corrdiff/generate.py]: updated

File: examples/weather/corrdiff/distill.py [hunk 5]
```diff
@@ -0,0 +1,1149 @@
...
+        regression_net=regression_net,
+        hr_mean_conditioning=cfg.model.hr_mean_conditioning,
+    )
+
+    # Instantiate the FastGenNet for distillation
+    model_cfg_update = DictConfig(
+        {
+            "precision": PRECISION_MAP[fp_optimizations],
+            "precision_infer": PRECISION_MAP[str(input_dtype)],
+            "input_shape": (diffusion_model.img_out_channels, *super_patch_shape),
```
[the remark this answers is not in the document]
Comment (jialusui1102) [reply examples/weather/corrdiff/distill.py:477]: added

File: examples/weather/corrdiff/distill.py [hunk 6]
```diff
@@ -0,0 +1,1149 @@
...
+
+def profiler_emit_nvtx():
+    if torch.cuda.is_available():
+        return torch.autograd.profiler.emit_nvtx()
+    else:
+        return nullcontext()
+
+
+# Train the CorrDiff model using the configurations in "conf/config_training.yaml"
+@hydra.main(version_base="1.2", config_path="conf", config_name="config_distill_mini")
```
[the remark this answers is not in the document]
Comment (jialusui1102) [reply examples/weather/corrdiff/distill.py]: updated
